src/main/kotlin/net/ccbluex/liquidbounce/utils/inventory/InventoryManager.kt

```diff
@@ -486,12 +488,23 @@ data class InventoryActionChain(
     val inventoryConstraints: InventoryConstraints,
     val actions: Array<out InventoryAction>,
     val priority: Priority
-) {
+) : Comparable<InventoryActionChain> {
 
     fun canPerformAction(): Boolean {
         return actions.all { action -> action.canPerformAction(inventoryConstraints) }
     }
 
-    fun requiresInventoryOpen() = actions.filterIsInstance<ClickInventoryAction>().any { it.screen == null }
+    fun requiresInventoryOpen() = actions.any { it is ClickInventoryAction && it.screen == null }
+
+    override fun compareTo(other: InventoryActionChain): Int {
+        // inventoryOpen asc (false -> true)
+        val thisOpenInv = requiresInventoryOpen()
+        if (thisOpenInv != other.requiresInventoryOpen()) {
+            return if (thisOpenInv) 1 else -1
+        }
+
+        // priority desc
+        return other.priority.compareTo(priority)
+    }
```

Comment:
Won't this basically eliminate the general priority as soon `inventoryOpen` is the case?